src/hooks/useSearchQuery.js

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+import { buildQueryParams, buildQuery } from './utils';
+
+const useSearchQuery = ({ text, relatedToVideoId, typeResult }) => {
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const queryParams = buildQueryParams({
+    part: 'snippet',
+    maxResults: 20,
+    ...(text && { q: text }),
+    type: typeResult || 'video',
+    key: process.env.REACT_APP_YOUTUBE_API_KEY,
+    ...(relatedToVideoId && { relatedToVideoId }),
+  });
+
+  const YOUTUBE_API_URL = buildQuery('search', queryParams);
```

Comment:
This is a great custom hook!
A slight problem is that it is consuming the API too fast when the user searches for something, this may have some impact on the performance or even the budget (if the API charges per request for example). There are a couple ways to optimize this, keeping the approach to search automatically a great solution is to use the `debounce()` method as discussed in the previous workshop 😉

Reply:
Yeah, I agree with you. I changed the event and I've add a button type submit to handle the request.